src/components/childComponents/CreateAndUpload.js

```diff
@@ -100,47 +167,142 @@ export default function CreateAndUpload() {
 			const body = {"name": name, "description": description};
 			const dataset = await createDatasetRequest(body);
 			if (dataset["id"] !== undefined) {
+				setClowderDataset(dataset);
 				await uploadToDatasetRequest(dataset["id"], dropFile).then((response) => {setClowderFile(response)} );
 			}
 		}
-		else {
-			console.log("error in dropped file");
-		}
 	}, [dropFile]);
 
-	// if clowderFile state has changed, submit file for extraction. if disabled by default.
+	// if clowderFile state has changed, submit file for extraction and preview html.
 	useEffect(async () => {
 		if (clowderFile !== null) {
 			const body = {"extractor": "ncsa.rctTransparencyExtractor"};
 			//const extractor_name = "ncsa.wordcount";
-			await extractionsRequest(clowderFile, body);
-
+			await extractionsRequest(clowderFile, body).then((response)=> setExtractionJob(response)); // end of extraction request
 		}
 	}, [clowderFile]);
 
+	// check extraction status in loop
+	useEffect( () => {
+		const loop = async () => {
+			const extraction_status = await checkExtractionStatus(clowderFile);
+			console.log(extraction_status);
+			if (extraction_status["Status"] === "Done" && extraction_status["ncsa.rctTransparencyExtractor"] === "DONE") {
+				const htmlFile = await checkHtmlInDataset(clowderDataset);
+				console.log(htmlFile);
+				if (typeof htmlFile.id === "string") {
+					// {"id":string, "size":string, "date-created":string, "contentType":text/html, "filename":string}
+					const previews_list = await getPreviews(htmlFile.id);
+					const preview = previews_list[0];
+					console.log(preview);
+					if (preview !== undefined) {
+						let previewsTemp = [];
+						// get all preview resources
+						let preview_config = {};
+						preview_config.previewType = preview["p_id"].replace(" ", "-").toLowerCase(); // html
+						preview_config.url = `${config.hostname}${preview["pv_route"]}?superAdmin=true`;
```

Comment:
`superAdmin=true` should not be necessary since the user is the creator of the resource and it's not a given that they will be an admin of the system.

Reply:
removed superAdmin query from url